### DISCUSSION TITLE
Unity auth with Solana wallet creation

### DISCUSSION THREAD

**Discussion by ericroy** [Web3Auth/Web3Auth#608] [Help]
Hi,

I'm integrating Web3Auth login in a Unity application.  The login flow is succeeding, but it is not automatically creating a Solana wallet for the newly logged in user.  The returned idToken contains one wallet of type `web3auth_app_key`, but I don't see one with type `solana`, which is what I was expecting.

In the Web3Auth Dashboard, I do have the app configured to use the Solana blockchain.

I notice in the documentation for the Javascript SDK, that you can provide a `chainConfig` property that specifies the type of wallet you want created.  I don't see anything equivalent in the Unity SDK.

Is it possible to have a Solana wallet automatically created during first login, using the Unity SDK?

Thanks!

**Comment by chaitanyapotti** [comment]
While initiating Unity sdk, are you setting `curve: "ed25519"` in login params? please do so

**Reply by ericroy** [reply]
Indeed, I was not setting the curve.  When I set it to Curve.ED25519, I still get the same results.  A JWT with just one entry in the wallets list:
```json
"wallets": [
    {
      "public_key": "036c9028577674...",
      "type": "web3auth_app_key",
      "curve": "secp256k1"
    }
  ],
```

I'm logging into an account that was previously created without this curve property set.  Could this be the problem?  Are wallets only created on first login (during account creation)?  If so, is it possible to delete a web3auth account, so I can login fresh for the first time?

**Reply by ericroy** [reply]
Tried logging in again with a whole different google account, this time the web3auth_app_key has the curve property I specified:
```json
"wallets": [
    {
      "public_key": "6fe153a82fd6e...",
      "type": "web3auth_app_key",
      "curve": "ed25519"
    }
  ],
```

But there is still no wallet with type `solana`.

I've read elsewhere in the github discussions that it's not possible to delete Web3Auth accounts.  That seems like rather a large problem for development/testing...  I'm running out of google accounts pretty fast here.

Anyway - any other thoughts on how to have a Solana wallet auto-created using the Unity sdk?  Thanks!

**Reply by ericroy** [reply]
~~Ok got it I think, with curve set to ed25519, I'm able to use the pubkey as a solana wallet, so long as I convert the hex to a base58 representation.  Thanks for your help!~~

Nevermind - I still don't think this is right.

Should I get a wallet with type "solana" in the the idToken, or not??

**Reply by himanshuchawla009** [reply]
"ed25519" is the curve used by solana to create accounts, so the public_key in array above is for solana only.

**Reply by ericroy** [reply]
Ok, I think I understand now, thanks
